databricks/koalas/indexing.py

```diff
@@ -181,6 +182,103 @@ def __getitem__(self, key):
                           or len(values) > 1) else values[0]
 
 
+class iAtIndexer(object):
+    """
+    Access a single value for a row/column pair by integer position.
+
+    Similar to ``iloc``, in that both provide integer-based lookups. Use
+    ``iat`` if you only need to get or set a single value in a DataFrame
+    or Series.
+
+    Raises
+    ------
+    KeyError
+        When label does not exist in DataFrame
+
+    Examples
+    --------
+    >>> df = ks.DataFrame([[0, 2, 3], [0, 4, 1], [10, 20, 30]],
+    ...                   columns=['A', 'B', 'C'])
+    >>> df
+        A   B   C
+    0   0   2   3
+    1   0   4   1
+    2  10  20  30
+
+    Get value at specified row/column pair
+
+    >>> df.iat[1, 2]
+    1
+
+    Get value within a series
+
+    >>> kser = ks.Series([1, 2, 3], index=[10, 20, 30])
+    >>> kser
+    10    1
+    20    2
+    30    3
+    Name: 0, dtype: int64
+
+    >>> kser.iat[1]
+    2
+    """
+    def __init__(self, kdf_or_kser):
+        from databricks.koalas.frame import DataFrame
+        from databricks.koalas.series import Series
+        assert isinstance(kdf_or_kser, (DataFrame, Series)), \
+            'unexpected argument type: {}'.format(type(kdf_or_kser))
+        self._kdf_or_kser = kdf_or_kser
+
+    @property
+    def _is_df(self):
+        from databricks.koalas.frame import DataFrame
+        return isinstance(self._kdf_or_kser, DataFrame)
+
+    @property
+    def _is_series(self):
+        from databricks.koalas.series import Series
+        return isinstance(self._kdf_or_kser, Series)
+
+    @property
+    def _internal(self):
+        return self._kdf_or_kser._internal
+
+    def __getitem__(self, key):
+        if self._is_df:
+            if not isinstance(key, tuple) or len(key) != 2:
+                raise TypeError(
+                    "Use DataFrame.iat like .iat[row_integer_position, column_integer_position]")
+            row_sel, col_sel = key
+        else:
+            assert self._is_series, type(self._kdf_or_kser)
+            if not isinstance(key, int) and len(key) != 1:
+                raise TypeError("Use Series.iat like .iat[row_integer_position]")
+            row_sel = key
+            col_sel = 0
+
+        if len(self._internal.index_map) == 1:
+            if is_list_like(row_sel):
+                raise ValueError(
+                    'iAt based indexing on a single index can only have a single value')
+        if not isinstance(col_sel, int):
+            raise ValueError('iAt based indexing on multi-index can only have integer values')
+        if isinstance(col_sel, int):
+            if col_sel > len(self._internal.data_columns):
+                raise KeyError(name_like_string(col_sel))
+            col_sel = (self._internal.data_columns[col_sel],)
+
+        sdf = self._internal.sdf.select(self._internal.data_columns)
+
```

Comment:
Should we retain the original index ? 
IMO code here discard the original index and attach a new `distributed-sequence` one. 

For example, If the index type is `distributed-sequence`, the sdf might looks like that. 

(SPARK_PARTITION_ID is the spark partition number, I create this spark.dataframe by creating a rdd and do repartition into 2)
```
+-----------------+--------------------+-----+
|__index_level_0__|SPARK_PARTITION_ID()|value|
+-----------------+--------------------+-----+
|                0|                   0|    2|
|                1|                   0|    6|
|                2|                   0|    7|
|       8589934592|                   1|    1|
|       8589934593|                   1|    3|
|       8589934594|                   1|    4|
|       8589934595|                   1|    5|
|       8589934596|                   1|    8|
+-----------------+--------------------+-----+
```

If I was the user, when I want to select value of `Row(8589934592,1,1)`, I would do `df.iat[8589934592, 2]` rather than `df.iat[3, 2]`. Because in distributed scenario the order is not important (and does not guarantee continuous) , I would directly use the index I see which is `8589934592`.

Reply:
hello @guyao ,

Thanks for your kindness review :)

maybe in that case what you're considering about,

i think we can use `AtIndexer` rather than `iAtIndexer` like below.

```python
>>> kdf  # koalas DataFrame
            value
0               2
1               6
2               7
8589934592      1
8589934593      3
8589934594      4
8589934595      5
8589934596      8
>>>
>>> kdf.at[8589934593, 'value']
3
```

Above all, the more fundamental reason is that we need to follow pandas' behavior like below.

```python
>>> pdf  # pandas DataFrame
            value
0               2
1               6
2               7
8589934592      1
8589934593      3
8589934594      4
8589934595      5
8589934596      8
>>> pdf.iat[4, 0]  # pandas `iat`, we should follow this behavior.
3

>>> kdf  # koalas DataFrame, as same as pandas' above
            value
0               2
1               6
2               7
8589934592      1
8589934593      3
8589934594      4
8589934595      5
8589934596      8
>>> kdf.iat[4, 0]  # koalas `iat` which should work same with pandas' `iat`
3
>>> kdf.at[]
```

Reply:
Got it, make sense. 
Thanks for explanation.

Reply:
My concern is that I am not sure whether Spark will guarantee any order.